bioconda_utils/bioconductor_skeleton.py

```diff
@@ -813,8 +812,11 @@ def write_recipe(package, recipe_dir, config, force=False, bioc_version=None,
 
             # Install and clean up
             R CMD INSTALL --library=$PREFIX/lib/R/library --build $TARBALL
-            rm $TARBALL""")
+            rm -r $STAGING""")
         with open(os.path.join(recipe_dir, 'post-link.sh'), 'w') as fout:
             fout.write(dedent(post_link_template))
+        pre_unlink_template = "rm -r $PREFIX/lib/R/library/{0}\n".format(package)
```

Comment:
Does a removal of that directory suffice or does
```bash
R CMD INSTALL --library=$PREFIX/lib/R/library --build $TARBALL
```
add or modify any other files?
(I know next to nothing about R, so someone more knowledgeable than me should be review this 😉)

Reply:
I did a quick test. The only files added by `R CMD INSTALL` are in `lib/R/library`. There are other files left after removing the package, but they are all created by conda.

```
$ conda create -y -n test bioconductor-genomeinfodbdata
$ find /opt/conda/ -iname *genomeinfodbdata*
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0.tar.bz2
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0/bin/.bioconductor-genomeinfodbdata-post-link.sh
/opt/conda/envs/test/conda-meta/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0.json
/opt/conda/envs/test/share/bioconductor-genomeinfodbdata-0.99.1-0
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/help/GenomeInfoDbData.rdb
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/help/GenomeInfoDbData.rdx
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/scripts/updateGenomeInfoDbData.R
/opt/conda/envs/test/bin/GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz
/opt/conda/envs/test/bin/.bioconductor-genomeinfodbdata-post-link.sh
$ conda remove -y -n test bioconductor-genomeinfodbdata
$ find /opt/conda/ -iname *genomeinfodbdata*
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0.tar.bz2
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0/bin/.bioconductor-genomeinfodbdata-post-link.sh
/opt/conda/envs/test/share/bioconductor-genomeinfodbdata-0.99.1-0
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/help/GenomeInfoDbData.rdb
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/help/GenomeInfoDbData.rdx
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/scripts/updateGenomeInfoDbData.R
/opt/conda/envs/test/bin/GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz
```

The files in `pkgs/` are also left behind when removing a normal R recipe:

```
$ conda install -y -n test r-registry
$ find /opt/conda/ -iname registry
/opt/conda/pkgs/r-registry-0.3-r3.4.1_0/lib/R/library/registry
/opt/conda/pkgs/r-registry-0.3-r3.4.1_0/lib/R/library/registry/R/registry
/opt/conda/envs/test/lib/R/library/registry
/opt/conda/envs/test/lib/R/library/registry/R/registry
$ conda remove -y -n test r-registry
$ find /opt/conda/ -iname registry
/opt/conda/pkgs/r-registry-0.3-r3.4.1_0/lib/R/library/registry
/opt/conda/pkgs/r-registry-0.3-r3.4.1_0/lib/R/library/registry/R/registry
```

Reply:
The `pkgs` folder is just Conda's package cache, no worries about that. But
```
/opt/conda/envs/test/share/bioconductor-genomeinfodbdata-0.99.1-0
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/help/GenomeInfoDbData.rdb
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/help/GenomeInfoDbData.rdx
/opt/conda/envs/test/lib/R/library/GenomeInfoDbData/scripts/updateGenomeInfoDbData.R
/opt/conda/envs/test/bin/GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz
```
are not created by `conda` itself but likely by `R`.
Does `R` have an uninstall command, i.e., some direct pendant to `R CMD INSTALL`?

Reply:
> Does R have an uninstall command, i.e., some direct pendant to R CMD INSTALL?

Yes, it does. We could switch to it, but it has the same effect in this case as the current `rm -r` (i.e. remove the directory in `lib/R/library`):

```
$ conda create -y -n test bioconductor-genomeinfodbdata
$ source activate test
$ R CMD REMOVE --library=/opt/conda/envs/test/lib/R/library/ GenomeInfoDbData
$ find /opt/conda/ -iname *genomeinfodbdata*
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0.tar.bz2
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0
/opt/conda/pkgs/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0/bin/.bioconductor-genomeinfodbdata-post-link.sh
/opt/conda/envs/test/conda-meta/bioconductor-genomeinfodbdata-0.99.1-r3.4.1_0.json
/opt/conda/envs/test/share/bioconductor-genomeinfodbdata-0.99.1-0
/opt/conda/envs/test/bin/GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz
/opt/conda/envs/test/bin/.bioconductor-genomeinfodbdata-post-link.sh
```

The directory in `share` is the directory `$STAGING` created by `post-link.sh` (which as we discussed above, this PR also proposes removing) and the files in `bin/` are added by conda.

Reply:
I'd go for `R CMD REMOVE`, but only because I don't know what `R`'s `INSTALL`/`REMOVE` can actually do..
So regardless if we use one or the other removal command, for the `GenomeInfoDbData` example only `$PREFIX/bin/GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz` will remain when this PR is adopted. This looks like some compilation file from `R` (who knows why the heck this ends up in `bin`...). To confirm this, `diff`ed the environment pre-install, post-install and post-removal:
```
$ diff -rqx conda-meta init/ installed/
Only in installed/bin: .bioconductor-genomeinfodbdata-post-link.sh
Only in installed/bin: GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz
Only in installed/lib/R/library: GenomeInfoDbData
Only in installed/share: bioconductor-genomeinfodbdata-0.99.1-0
$ diff -rqx conda-meta installed/ removed/
Only in installed/bin: .bioconductor-genomeinfodbdata-post-link.sh
$ diff -rqx conda-meta init/ removed/
Only in removed/bin: GenomeInfoDbData_0.99.1_R_x86_64-pc-linux-gnu.tar.gz
Only in removed/lib/R/library: GenomeInfoDbData
Only in removed/share: bioconductor-genomeinfodbdata-0.99.1-0
```

Reply:
Ah ok, IIUC, `R CMD INSTALL --build` installs the package first and then creates the tarball in `bin`. Since I don't believe we need/want that tarball anyway, shouldn't the command just be
```bash
R CMD INSTALL --library=$PREFIX/lib/R/library $TARBALL
```
without the `--build`?